app/jobs/plexer_job.rb

```diff
@@ -13,10 +13,10 @@
 # Do not assume we can just get metadata from (the DruidVersionZip) zip.
 # Jobs are not run at the same time or on the same system, so the info may not match.
 # Therefore, we receive the info passed by the process that was there when the file was created.
-class PlexerJob < DruidVersionJobBase
+class PlexerJob < ZipPartJobBase
   queue_as :zips_made
 
-  before_enqueue { |job| job.zip_info_check!(job.arguments.third) }
+  before_enqueue { |job| job.zip_info_check!(job.arguments.fourth) }
```

Comment:
The way I read this line in context, you can see from the yard docs for the `perform` method that the metadata is fourth. The same goes for me with 'S3WestDeliveryJob'. I'd argue that the yard doc there is our specification already.

Reply:
i think that would make it clearer to me. +1 @ndushay !

Reply:
A hash for all the args is a bad idea because it breaks queue locking semantics.  The proposed method(s) would only be called once per execution, so passing all the args to get back one doesn't add anything.  The check is implemented in the `before_enqueue` callback.  Literally any logic could be used there, and referencing args (by position) is commonplace.